test/e2e/scaffold/scaffold.go

```diff
@@ -426,30 +426,16 @@ func (s *Scaffold) beforeEach() {
 func (s *Scaffold) afterEach() {
 	defer ginkgo.GinkgoRecover()
 
-	if ginkgo.CurrentSpecReport().Failed() {
-		_, _ = fmt.Fprintln(ginkgo.GinkgoWriter, "Dumping namespace contents")
-		output, _ := k8s.RunKubectlAndGetOutputE(ginkgo.GinkgoT(), s.kubectlOptions, "get", "deploy,sts,svc,pods")
-		if output != "" {
-			_, _ = fmt.Fprintln(ginkgo.GinkgoWriter, output)
-		}
-		output, _ = k8s.RunKubectlAndGetOutputE(ginkgo.GinkgoT(), s.kubectlOptions, "describe", "pods")
-		if output != "" {
-			_, _ = fmt.Fprintln(ginkgo.GinkgoWriter, output)
-		}
-		// Get the logs of apisix
-		output = s.GetDeploymentLogs("apisix-deployment-e2e-test")
-		if output != "" {
-			_, _ = fmt.Fprintln(ginkgo.GinkgoWriter, output)
-		}
-		// Get the logs of ingress
-		output = s.GetDeploymentLogs("ingress-apisix-controller-deployment-e2e-test")
-		if output != "" {
-			_, _ = fmt.Fprintln(ginkgo.GinkgoWriter, output)
-		}
+	shouldDeleteNamespace := true
+	if ginkgo.CurrentSpecReport().Failed() && os.Getenv("E2E_ENV") == "dev" {
+		// Keep namespace if test failed in dev mode
+		shouldDeleteNamespace = false
```

Comment:
We are not going to remove the dump feature cuz we need it to debug CI failures.
I mean, if the environment is dev, we can skip dump. When we are in CI environment, we still need dump.

Reply:
Sure, my bad. Of course we need dump namespace info in CI. Thanks for your review!